### ISSUE TITLE
[Question] Provider not working correctly?

### ISSUE THREAD

**Issue by cloeper** [jfairbank/redux-saga-test-plan#292]
Hey folks! Trying to figure out if I'm using providers wrong, they aren't working correctly, or I'm just doing something that simply isn't possible. Any help is appreciated.

**The Problem**
I expect the provider to use the fake data I'm providing, and that the API call is _not_ actually made.

What's actually happening is that the API is made and the fake data in the provider is disregarded.

Here's my code:
----------
I have a file that has my axios api calls. Like so:
```
./Login.api.ts

import axios from '../axios';

export async function postLoginInstanceMeta(email: string) {
  return axios.post(`/api/login/instance/`, { email });
}
```

Then in my saga, I'm calling it like this:
```
./Saga.ts

const result = yield call(postLoginInstanceMeta, email);
```

And here's how I'm attempting to test it:
```
it('on success sets the login phase correctly when the user enters a non-SSO email address', () => {
  const email = 'email@address.com';
  const nonSSOResponse: ILoginInstanceMetaResponse = {
      ...data
  };

  return expectSaga(checkInstance, instanceCheckActionCreators.checkInstance(email))
    .provide([[call(postLoginInstanceMeta, 'email@address.com'), nonSSOResponse]])
    .put(loginActionCreators.setLoginPhase(LoginPhases.PasswordLogin))
    .call(redirect, `foo.localhost:3001/login?t=1234`)
    .silentRun();
});
```

What am I missing here? The docs seem to indicate that this should work the way I expect it to.

Thanks!

**Comment by srinivasadnt3** [comment]
I am also facing the same issue

`    return expectSaga(x, {})
      .provide([[call(axios.get, ''), responseData]])
      .put({
        type: type:SOMEACTION,
        payload: responseData
      })
      .dispatch({ type:SOMEACTION })
      .silentRun();`
It seems like this issue is open since quite long, I would request to provide any solution for this.

For now to resolve this i have manually mocked the service and its working fine for me but i am pretty sure that its not the right way

`    jest.mock('axios');
    axiosMock.get.mockResolvedValueOnce({
      data: responseData
    });
    return expectSaga(x, {})
      .provide([[call(axios.get, ''), responseData]])
      .put({
        type: ORDER_SEARCH_PAGE_CONTENT_LOADED_ACTION_TYPE,
        payload: responseData
      })
      .dispatch({ type: FETCH_ORDER_SEARCH_PAGE_CONTENT_ACTION_TYPE })
      .silentRun();`

**Comment by srinivasadnt3** [comment]
I am also facing the same issue

`    return expectSaga(x, {})
      .provide([[call(axios.get, ''), responseData]])
      .put({
        type: type:SOMEACTION,
        payload: responseData
      })
      .dispatch({ type:SOMEACTION })
      .silentRun();`
It seems like this issue is open since quite long, I would request to provide any solution for this.

For now to resolve this i have manually mocked the service and its working fine for me but i am pretty sure that its not the right way

`    jest.mock('axios');
    axiosMock.get.mockResolvedValueOnce({
      data: responseData
    });
    return expectSaga(x, {})
      .provide([[call(axios.get, ''), responseData]])
      .put({
        type: ORDER_SEARCH_PAGE_CONTENT_LOADED_ACTION_TYPE,
        payload: responseData
      })
      .dispatch({ type: FETCH_ORDER_SEARCH_PAGE_CONTENT_ACTION_TYPE })
      .silentRun();`

**Comment by srinivasadnt3** [comment]
I am also facing the same issue

```    
      return expectSaga(x, {})
      .provide([[call(axios.get, ''), responseData]])
      .put({
        type: type:SOMEACTION,
        payload: responseData
      })
      .dispatch({ type:SOMEACTION })
      .silentRun();
````
It seems like this issue is open since quite long, I would request to provide any solution for this.

For now to resolve this i have manually mocked the service and its working fine for me but i am pretty sure that its not the right way

```
    jest.mock('axios');
     // instead of provide[some data] we could follow below mocking
    axiosMock.get.mockResolvedValueOnce({
      data: responseData
    });
    return expectSaga(x, {})
      .put({
        type: ORDER_SEARCH_PAGE_CONTENT_LOADED_ACTION_TYPE,
        payload: responseData
      })
      .dispatch({ type: FETCH_ORDER_SEARCH_PAGE_CONTENT_ACTION_TYPE })
      .silentRun();
```

**Comment by Jimit09** [comment]
Yeah. It doesn't seem to work. Even I am facing issue. Unable to mock using provider.

**Comment by WillyRamirez** [comment]
I am able to mock API calls like this:

`[matchers.call.fn(api.getReservation), responses.successful(getReservationResponses.RESPONSES.DEFAULT)]`

**Comment by aprisniak** [comment]
I am facing issue. Unable to mock using provider.
